pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -118,10 +119,10 @@ func (e *CloudWatchExecutor) executeTimeSeriesQuery(ctx context.Context, queryCo
 		}
 
 		if query.Id == "" && query.Expression != "" {
-			result.Results[query.RefId] = &tsdb.QueryResult{
+			results.Results[query.RefId] = &tsdb.QueryResult{
```

Comment:
Possible to use same solution as sql datasources [here](https://github.com/grafana/grafana/blob/master/pkg/tsdb/sql_engine.go#L131-L132) and then provide needed arguments as in-parameters to go routine? 

Or is that not possible due to use off errgroup?

Reply:
Your approach looks nice, and I want to use it.
But, I consider the code becomes little complicated, because `executeGetMetricDataQuery()` will response multiple `tsdb.QueryResult` in one go routine, and it is called for each regions.
https://github.com/grafana/grafana/blob/v5.3.1/pkg/tsdb/cloudwatch/cloudwatch.go#L147

I need to pass the value typed like `map[string]*tsdb.QueryResult` to `executeGetMetricDataQuery()`.
There are multiple values `map[string]*tsdb.QueryResult` for each regions, so I need to keep it until query finish, and copy to `result.Results`.

If coding policy allows I want to use channel for avoding concurrent map writes.

Reply:
@marefr I success to reproduce the issue.
https://gist.github.com/mtanda/954793aa2ff66b5ffbc782059fde01ac

I add template variable for InstanceId, which include 50+ instance ids, and make EC2 CPUUtilization graph for the variable.

I test my fix doesn't cause the issue. refresh every 5sec, and wait for 10 minutes, the Grafana doesn't crash.

Reply:
Okay. I only have one instance id available so hard for me to test. But I'll approve this and merge.